lib/make-badge.js

```diff
@@ -119,16 +117,12 @@ function makeBadge (measurer, {
     text = text.map(value => value.toUpperCase());
   }
 
-  if (colorscheme) {
-    let pickedColorscheme = definedColorschemes[colorscheme];
-    if (!pickedColorscheme) {
-      pickedColorscheme = definedColorschemes.red;
-    }
-    colorA = colorA || pickedColorscheme.colorA;
-    colorB = colorB || pickedColorscheme.colorB;
-  }
-  if (!cssColor.test(colorA)) { colorA = undefined; }
-  if (!cssColor.test(colorB)) { colorB = undefined; }
+  // colorA/B have a higher priority than colorscheme
+  colorA = colorA || colorscheme || undefined;
+  colorB = colorB || colorscheme || undefined;
+  colorA = definedColorschemes[colorA] !== undefined ? definedColorschemes[colorA].colorA || undefined : validColor(colorA) ? colorA : undefined;
+  colorB = definedColorschemes[colorB] !== undefined ? definedColorschemes[colorB].colorB || undefined : validColor(colorB) ? colorB : undefined;
+  // check if colorA/B is a colorscheme else check if it's a valid css3 color else return undefined and let the badge assign the default color
```

Comment:
These are a bit tricky to read! Thanks for the comment, though I think better to replace it with code that's more readable. How about adding a helper function with an `if/else if/else`?